rest-server/src/test/java/org/opennms/horizon/server/service/flows/UnitConverterTest.java

```diff
@@ -0,0 +1,122 @@
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.horizon.server.service.flows;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.opennms.horizon.server.model.flows.FlowingPoint;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+class UnitConverterTest {
+    private static final String INGRESS = "INGRESS";
+    private static final String EGRESS = "EGRESS";
+    private static final String LABEL = "LABEL";
+
+    @Test
+    void testSingleConvert() {
+        List<FlowingPoint> pointList = new ArrayList<>();
+
+        long startTime = 1694473114833L;
+        pointList.add(getFlowingPoint(startTime, 100, LABEL, INGRESS));
+        pointList.add(getFlowingPoint(startTime + 100, 200, LABEL, INGRESS));
+        pointList.add(getFlowingPoint(startTime + 200, 300, LABEL, INGRESS));
+        pointList.add(getFlowingPoint(startTime + 300, 400, LABEL, INGRESS));
+        var output = UnitConverter.convertToRate(pointList);
+
+        Assertions.assertEquals(4, output.size());
+        Assertions.assertEquals(8000, output.get(0).getValue());
+        Assertions.assertEquals(16_000, output.get(1).getValue());
+        Assertions.assertEquals(24_000, output.get(2).getValue());
+        Assertions.assertEquals(32_000, output.get(3).getValue());
+        for (int i = 0; i < pointList.size(); i++) {
+            Assertions.assertEquals(startTime + 100L * i, pointList.get(i).getTimestamp().toEpochMilli());
+            Assertions.assertEquals(INGRESS, pointList.get(i).getDirection());
+        }
+    }
+
+
+    @Test
+    void testInvalidList() {
+        List<FlowingPoint> pointList = new ArrayList<>();
+        pointList = UnitConverter.convertToRate(pointList);
+        Assertions.assertEquals(0, pointList.size());
+
+        // single data
+        pointList.add(getFlowingPoint(System.currentTimeMillis(), 100, LABEL, INGRESS));
+        pointList = UnitConverter.convertToRate(pointList);
+        Assertions.assertEquals(0, pointList.size());
+
+        // null
+        Assertions.assertEquals(0,  UnitConverter.convertToRate(null).size());
+    }
+
+    @Test
+    void testConvert() {
+        List<FlowingPoint> pointList = new ArrayList<>();
+
+        long startTime = 1694473114833L;
+        pointList.add(getFlowingPoint(startTime, 100, LABEL, INGRESS));
+        pointList.add(getFlowingPoint(startTime + 100, 200, LABEL, INGRESS));
+        pointList.add(getFlowingPoint(startTime + 200, 300, LABEL, INGRESS));
+        pointList.add(getFlowingPoint(startTime + 300, 400, LABEL, INGRESS));
+
+        pointList.add(getFlowingPoint(startTime, 100, LABEL + "1", INGRESS));
+        pointList.add(getFlowingPoint(startTime, 100, LABEL, EGRESS));
+        pointList.add(getFlowingPoint(startTime + 100, 200, LABEL, EGRESS));
+
+        Collections.shuffle(pointList);
```

Comment:
I think it would be better to have a pre-defined, asorted order instead of randomly shuffling during test execution.

```suggestion
        pointList.add(getFlowingPoint(startTime + 100, 200, LABEL, EGRESS));
        pointList.add(getFlowingPoint(startTime, 100, LABEL, INGRESS));
        pointList.add(getFlowingPoint(startTime + 300, 400, LABEL, INGRESS));
        pointList.add(getFlowingPoint(startTime + 100, 200, LABEL, INGRESS));

        pointList.add(getFlowingPoint(startTime, 100, LABEL + "1", 
        pointList.add(getFlowingPoint(startTime + 200, 300, LABEL, INGRESS));
        pointList.add(getFlowingPoint(startTime, 100, LABEL, EGRESS));
```

Reply:
The main idea is to maintain the test data order for easier maintenance. replace with add with index.